src/proxy/http/insert.rs

```diff
@@ -44,16 +44,26 @@ pub struct ValLazy<V>(V);
 
 pub fn layer<F, V>(f: F) -> Layer<FnLazy<F>, V>
 where
-    F: Fn() -> V,
+    F: Fn() -> V + Clone,
```

Comment:
"compiler said so" --- the `new` function I added has a `Lazy` bound:
https://github.com/linkerd/linkerd2-proxy/blob/0231f60979e9d988635c1f0f90b2a72a677681c3/src/proxy/http/insert.rs#L55-L57
and `FnLazy<F>` is only `Lazy` when `F: Clone`:
https://github.com/linkerd/linkerd2-proxy/blob/0231f60979e9d988635c1f0f90b2a72a677681c3/src/proxy/http/insert.rs#L168-L171

alternatively, we could get rid of the `new`, and not have this bound here, but it would allow constructing `Layer`s that don't _actually_ implement `Layer`, resulting in a much bigger & harder to debug type error later...